components/brave_wallet/browser/brave_wallet_provider.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_PROVIDER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_PROVIDER_H_
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+class BraveWalletService;
+
+namespace brave_wallet {
+
+class BraveWalletProvider final
```

Comment:
this should be `BraveWalletProviderImpl` to distinguish it from the interface. Also the general convention for naming these seems to be *Host to BraveWalletHost?

Reply:
renamed